pkg/wal/processor/transformer/wal_transformer_validator.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package transformer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v5/pgtype"
+	pglib "github.com/xataio/pgstream/internal/postgres"
+	"github.com/xataio/pgstream/pkg/transformers"
+	"golang.org/x/exp/slices"
+)
+
+func GetTransformerValidationFunc(ctx context.Context, pgurl, snapshoturl string) func(*Transformer) error {
+	// if pgurl or snapshot url is provided, use postgres validator
+	if pgurl == "" {
+		if snapshoturl == "" {
+			return nil
+		}
+		pgurl = snapshoturl
+	}
+	return pgTransformerValidator(ctx, pgurl)
+}
+
+func pgTransformerValidator(ctx context.Context, pgurl string) func(t *Transformer) error {
+	return func(t *Transformer) error {
+		conn, err := pglib.NewConn(ctx, pgurl)
+		if err != nil {
+			return fmt.Errorf("creating postgres connection pool: %w", err)
```

Comment:
This is a single connection rather than a pool, so this can be updated to remove the pool part (it can be misleading otherwise).